app/graphql/mutations/update_sample_metadata.rb

```diff
@@ -6,17 +6,25 @@ class UpdateSampleMetadata < BaseMutation
     description 'Update metadata for a sample.'
     argument :metadata, GraphQL::Types::JSON, required: true, description: 'The metadata to update the sample with.'
     argument :sample_id, ID,
-             required: true,
-             description: 'The Node ID of the sample to be updated.'
+             required: false,
+             description: 'The Node ID of the sample to be updated. For example, `gid://irida/Sample/a84cd757-dedb-4c64-8b01-097020163077`' # rubocop:disable Layout/LineLength
+    argument :sample_puid, ID, # rubocop:disable GraphQL/ExtractInputType
+             required: false,
+             description: 'Persistent Unique Identifier of the sample. For example, `INXT_SAM_AAAAAAAAAA`.'
+    validates required: { one_of: %i[sample_id sample_puid] }
 
     field :errors, [String], null: false, description: 'A list of errors that prevented the mutation.'
     field :sample, Types::SampleType, null: false, description: 'The updated sample.'
     field :status, GraphQL::Types::JSON, null: false, description: 'The status of the mutation.'
 
-    def resolve(sample_id:, metadata:)
-      sample = IridaSchema.object_from_id(sample_id, { expected_type: Sample })
+    def resolve(args)
+      sample = if args[:sample_id]
+                 IridaSchema.object_from_id(args[:sample_id], { expected_type: Sample })
+               else
+                 Sample.find(args[:sample_puid])
```

Comment:
Same comment as above about using `find_by` instead of `find`

Reply:
Fixed in c362e774 good catch!